build_darwin.sh

```diff
@@ -61,11 +61,11 @@ cd tor-$TOR_VERSION && \
             --enable-static-libevent \
             --enable-static-openssl  \
             --with-libevent-dir=$PWD/../libevent-$LIBEVENT_VERSION/install \
-            --with-openssl-dir=$PWD/../openssl-$OPENSSL_VERSION/install \
+            --with-openssl-dir=$PWD/../openssl-$OPENSSL_VERSION/ \
             --disable-asciidoc \
             ac_cv_func_getentropy=no \
             ac_cv_func_clock_gettime=no && \
-make ${jobs:+-j${jobs}} && make ${jobs:+-j${jobs}} check && make install
+make ${jobs:+-j${jobs}} && make ${jobs:+-j${jobs}} check
 cd ..
 
-cp tor-$TOR_VERSION/install/bin/tor tor-$TOR_VERSION-darwin-brave-$BRAVE_TOR_VERSION
+cp tor-$TOR_VERSION/src/app/tor tor-$TOR_VERSION-darwin-brave-$BRAVE_TOR_VERSION
```

Comment:
This creates a file named tor-0.3.5.7-darwin-brave-0 (in my test environment), is that accurate? The README says "Increment the Brave version number for each published build.". If so this seems fine to me.

Reply:
We were initially updating the `Brave Build` as well, but I think it makes more sense to update the Brave build only if we release two updates for same tor version if there was an un-intended crash or bug.

Reply:
What I intended when I suggested a Brave revision number is to have it count up from zero independently in each version of tor, so we might have 0.3.5.7brave0, 0.3.5.7brave1, 0.3.5.7brave2, 0.3.5.9brave0, &c.

Reply:
How can we (and more to the point, should we) codify that so it works that way in Jenkins builds? It might be harder to track this value if it restarts at 0 for every version of Tor. An alternative is to use a Jenkins environment variable (typically the [${BUILD_NUMBER}](https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables)) which is guaranteed to never repeat, but always increment positively for every Jenkins build of the project.

Thoughts?

Reply:
We could do something like this in the Jenkins job:

```
[user@host tor_build_scripts]$ echo $BUILD_NUMBER
124
[user@host tor_build_scripts]$ sed -i "s/export BRAVE_TOR_VERSION=\"0\"/export BRAVE_TOR_VERSION=\"${BUILD_NUMBER}\"/" env.sh
[user@host tor_build_scripts]$ git diff env.sh
diff --git a/env.sh b/env.sh
index cb4f102..ee8aff4 100644
--- a/env.sh
+++ b/env.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-export BRAVE_TOR_VERSION="0"
+export BRAVE_TOR_VERSION="124"

```

Reply:
I don't see any particular problem with using $BUILD_NUMBER.  I was just treating the revision number as an even more minor number, kind of like package managers have a package revision number.  But if $BUILD_NUMBER is more convenient, we could do that instead.

Hmm...  We _could_ also memorialize the build number of the last build whenever we bump the version, and then subtract that from $BUILD_NUMBER.  But I don't care enough about following the arbitrary rule I suggested a long time ago to make sure this works reliably.